File: java-code/src/main/java/refactoring/encapsulation/extract_class/after/TelephoneNumber.java
package refactoring.encapsulation.extract_class.after;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

//STEP 2: Create Telephone number class
@Getter
@Setter
@ToString
public class TelephoneNumber {
  private String number;

  //Step3: Move telephone number related fields to this class
  private int officeAreaCode;
  private int officeNumber;

  public String getNumber() {

    return this.officeAreaCode
        + " "
        + this.officeNumber
        + " "
        + this.number;
  }

}

File: go-code/organizing_data/change_reference_to_value/after/telephone-number.go
package after

// STEP1: struct fields are mutable by default. Lets try to make Telephone number immutable
type TelephoneNumber struct {
	areaCode string
	number   string
}

// Create a constructor function
func New(areaCode string, number string) TelephoneNumber {
	return TelephoneNumber{
		areaCode: areaCode,
		number:   number,
	}
}

func (tn TelephoneNumber) getAreaCode() string {
	return tn.areaCode
}

func (tn TelephoneNumber) getNumber() string {
	return tn.number
}
